subsys/net/l2/ieee802154/ieee802154_fragment.c

```diff
@@ -334,34 +345,23 @@ static inline struct frag_cache *get_reass_cache(u16_t size, u16_t tag)
 	return NULL;
 }
 
-/* Helper function to write fragment data to Rx packet based on offset. */
-static inline bool copy_frag(struct net_pkt *pkt,
-			     struct net_buf *frag,
-			     u16_t offset)
+/* Helper function to append fragment data to Rx packet based on offset. */
+static inline bool append_frag(struct net_pkt *pkt,
+			       struct net_buf *frag,
+			       u16_t offset)
 {
-	struct net_buf *input = frag;
-	struct net_buf *write;
-	u16_t pos = offset;
-
-	write = pkt->frags;
-
-	while (input) {
-		write = net_pkt_write(pkt, write, pos, &pos, input->len,
-				       input->data, NET_6LO_RX_PKT_TIMEOUT);
-		if (!write && pos == 0xffff) {
-			/* Free the new bufs we tried to get, we need to discard
-			 * the whole fragment chain.
-			 */
-			net_pkt_frag_unref(pkt->frags);
-			pkt->frags = NULL;
+	if (offset != net_pkt_get_len(pkt)) {
```

Comment:
Look at my previous comment. Sometime offset could be more than what currently received total packet length.